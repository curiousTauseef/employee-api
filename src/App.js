import React, { Component } from 'react'
import styled from 'styled-components'
import logo from './logo.svg'
import './App.css'

const Wrapper = styled.div`
    text-align: center;
`

class App extends Component {
    render() {
        return (
            <Wrapper>
                <header className="App-header">
                    <img src={logo} className="App-logo" alt="logo" />
                    <h1>
                        Edit <code>src/App.js</code> and save to reload.
                    </h1>
                    <a
                        className="App-link"
                        href="https://reactjs.org"
                        target="_blank"
                        rel="noopener noreferrer"
                    >
                        Learn React
                    </a>
                </header>
            </Wrapper>
        )
    }
}

export default App
