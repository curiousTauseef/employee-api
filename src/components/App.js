import React from 'react'
import styled, { ThemeProvider, keyframes } from 'styled-components'
import logo from '../img/logo.svg'
import Fetch from './Fetch'
import Employee from './Employee'
import Pagination from './Pagination'
import theme from '../theme'
import GlobalStyle from '../styles/GlobalStyle'

const Wrapper = styled.div`
    text-align: center;
`

const Header = styled.header`
    margin: 40px 0;
`

const List = styled.ul`
    list-style: none;
    padding: 0;
    margin: 0 20px 40px 20px;
    text-align: left;
    background-color: ${props => props.theme.white};
    box-shadow: ${props => props.theme.boxShadow};
`

const spinner = keyframes`
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
`

const Loading = styled.p`
    display: flex;
    align-items: center;
    justify-content: center;

    img {
        animation: ${spinner} infinite 2s linear;
        width: 50px;
        display: block;
    }
`

const Main = styled.main`
    max-width: 768px;
    margin: 0 auto;
`

const App = () => (
    <ThemeProvider theme={theme}>
        <Wrapper>
            <GlobalStyle />
            <Header>
                <h1>Employee List</h1>
            </Header>
            <Main>
                <Fetch endpoint="http://localhost:8000/api/employees">
                    {({ data, error, loading }) => {
                        if (loading)
                            return (
                                <Loading>
                                    <span>
                                        <img src={logo} alt="Spinner" />
                                    </span>
                                    <span>Loading...</span>
                                </Loading>
                            )
                        if (error) return <p>{error}</p>
                        return (
                            <div>
                                <Pagination data={data} perPage={10}>
                                    {({ pageData }) => (
                                        <List>
                                            {pageData.map(employee => (
                                                <Employee
                                                    key={employee.uuid}
                                                    employee={employee}
                                                />
                                            ))}
                                        </List>
                                    )}
                                </Pagination>
                            </div>
                        )
                    }}
                </Fetch>
            </Main>
        </Wrapper>
    </ThemeProvider>
)

export default App
