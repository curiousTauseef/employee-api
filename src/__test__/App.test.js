import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'

describe('<App/>', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<App />)
        const title = wrapper.find('h1')
        console.log(title.debug())
        expect(title.text()).toContain('Edit')
    })
})
