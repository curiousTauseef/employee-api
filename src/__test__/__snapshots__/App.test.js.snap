// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders without crashing 1`] = `
<ThemeProvider
  theme={
    Object {
      "black": "#000000",
      "blue": "#0073b1",
      "boxShadow": "0 0 0 1px rgba(0,0,0,.15), 0 2px 3px rgba(0,0,0,.2)",
      "grey": "#e2e2e2",
      "greyBg": "#f6f6f6",
      "greyTxt": "#888888",
      "white": "#ffffff",
    }
  }
>
  <styled.div>
    <GlobalStyleComponent />
    <styled.header>
      <h1>
        Employee List
      </h1>
    </styled.header>
    <styled.main>
      <Fetch
        endpoint="http://localhost:8000/api/employees"
      >
        <Component />
      </Fetch>
    </styled.main>
  </styled.div>
</ThemeProvider>
`;
